external/storm-hbase/src/main/java/org/apache/storm/hbase/common/Utils.java

```diff
@@ -47,14 +48,28 @@ public static HTable getTable(UserProvider provider, Configuration config, Strin
             ugi = UserGroupInformation.getCurrentUser();
 
             LOG.debug("UGI for current USER : {}", ugi.getUserName());
+            boolean foundHBaseAuthToken = false;
             for (Token<? extends TokenIdentifier> token : ugi.getTokens()) {
                 LOG.debug("Token in UGI (delegation token): {} / {}", token.toString(),
                         token.decodeIdentifier().getUser());
 
-                // use UGI from token
-                ugi = token.decodeIdentifier().getUser();
-                ugi.addToken(token);
+                // token.getKind() = Text, Text is annotated by @Stringable
+                // which ensures toString() implementation
+                if (token.getKind().toString().equals(TOKEN_KIND_HBASE_AUTH_TOKEN)) {
+                    // use UGI from token
+                    LOG.debug("Found HBASE_AUTH_TOKEN - using the token to replace current user.");
+
+                    ugi = token.decodeIdentifier().getUser();
+                    ugi.addToken(token);
+
+                    foundHBaseAuthToken = true;
```

Comment:
@HeartSaVioR Can there be more than one HBASE_AUTH_TOKEN token for the current UGI?

Reply:
I don't think if there's one HBase cluster needed to be connected in worker, but not sure if there're more than one HBase clusters. If we need to consider and handle multiple tokens, matching token would be completely different scenario.

Reply:
I am trying to understand the current limitations. Can a worker support two or more HBase bolts talk to respective HBase clusters using delegation tokens?

Reply:
I didn't test it actually, but looks like we don't support it, since we are only using **one** of tokens. Not sure HDFS/Hive bolt supports it, too.

Reply:
@HeartSaVioR There should not be multiple tokens with HBASE_AUTH_TOKEN key with the current implementation. It is better to log a message(atleast with info level) if we find multiple tokens so that it is easy to debug such scenarios.

Reply:
Makes sense. I'll update the patch.

Reply:
@satishd Addressed.